### ISSUE TITLE
Permission errors with clearing cache

### ISSUE THREAD

**Issue by ErikGartner** [AdmitHub/meteor-buildpack-horse#161]
##### Type of problem
 - Failure to build

##### Summary
The buildpack fails to delete the `/cache` folder when `BUILDPACK_CLEAR_CACHE=1`.

##### Contents of `.meteor/release`
```
METEOR@1.4.3.1
```

##### Buildpack version
ae206a7ef1da461f7f8cf0c7a16b77affb5019e9

##### Logs
```
Counting objects: 33, done.
Delta compression using up to 4 threads.
Compressing objects: 100% (28/28), done.
Writing objects: 100% (33/33), 667.42 KiB | 0 bytes/s, done.
Total 33 (delta 10), reused 0 (delta 0)
-----> Cleaning up...
-----> Building ghostdoc from herokuish...
-----> Adding BUILD_ENV to build environment...
-----> Fetching custom buildpack
-----> Node.js app detected
----> Clearing cache dir.
remote: rm: cannot remove '/cache': Permission denied
```

##### Deployment target
Dokku

Do you think this is a Dokku bug? Looking at `/home/dokku/<app>/` I get:

```
total 100
drwxr-xr-x 11 dokku dokku 4096 Mar  1 13:10 .
drwxr-xr-x 17 dokku dokku 4096 Mar  1 13:01 ..
drwxrwxr-x  2 dokku dokku 4096 Oct  7 15:05 branches
drwxrwxr-x  3 32767 32767 4096 Mar  1 13:10 cache
-rw-rw-r--  1 dokku dokku   66 Oct  7 15:05 config
-rw-rw-r--  1 dokku dokku   65 Feb 28 12:53 CONTAINER.web.1
-rw-rw-r--  1 dokku dokku    0 Mar  1 13:10 .deploy.lock
-rw-rw-r--  1 dokku dokku   73 Oct  7 15:05 description
-rw-r--r--  1 dokku dokku  104 Oct 14 21:33 DOCKER_OPTIONS_BUILD
-rw-r--r--  1 dokku dokku  128 Oct 14 21:33 DOCKER_OPTIONS_DEPLOY
-rw-r--r--  1 dokku dokku  104 Oct 14 21:33 DOCKER_OPTIONS_RUN
-rw-rw-r--  1 dokku dokku    6 Oct  7 15:09 DOKKU_SCALE
-rw-------  1 dokku dokku  446 Feb 22 00:01 ENV
-rw-rw-r--  1 dokku dokku   23 Oct  7 15:05 HEAD
...
```

(Quick off-topic question: Is it safe to delete the cache folder after the slug is built?)

**Comment by tab00** [comment]
Would it be better to do `rm -rfv $2/*`?

i.e. clear the contents instead of the actual directory.  That's how it's done in https://github.com/perplexes/heroku-buildpack-clearcache/blob/master/bin/compile#L5

**Comment by yourcelf** [comment]
Sure, that could work. Heroku seems to tolerate removing the cache directory that is passed as an argument.  A glob would fail to remove any hidden files, but our main interest is to remove files that we might have created, which would not be hidden.

The cache should be safe to delete from the perspective of meteor-buildpack-horse, but I don't know a lot about Dokku's expectations for its persistence (we test mainly with heroku).

A side question: why are you running with BUILDPACK_CLEAR_CACHE?  Is there a problem you're encountering when you leave the cache intact?

**Comment by tab00** [comment]
In my case I have this issue regarding cache which would require clearing it: https://github.com/AdmitHub/meteor-buildpack-horse/issues/162

**Comment by ErikGartner** [comment]
I want to clear the cache because my ~8 meteor apps take up about 12gb of disk space.

I've experimented with doing `rm -rfv $CACHE/*` but nothing seemed to get deleted and I didn't get any print outs from `-v`. Here is my [commit](https://github.com/ErikGartner/meteor-buildpack-horse/commit/adeba5a8929fdafb11eca0787c8e93101c84614c) that tries to do that.

**Comment by yourcelf** [comment]
Should be fixed in 3da0ca364d404b7ad162f80ab1b6ff6f20c358a4. Thanks, @ErikGartner!
